File: engine/src/main/java/pl/polsl/hdised/engine/measurement/MeasurementDto.java
package pl.polsl.hdised.engine.measurement;

import java.util.Date;

public class MeasurementDto {

    private String cityName;
    private Date date;
    private String deviceId;
    private Float temperature;
    private String unit;

    public Float getTemperature() {
        return temperature;
    }

    public String getUnit() {
        return unit;
    }

    public String getCityName() {
        return cityName;
    }

    public Date getDate() {
        return date;
    }

    public String getDeviceId() {
        return deviceId;
    }
}

File: golang-measurement-generator/measurementDto.go
package main

import "time"

type MeasurementDto struct {
	CityName    string    `json:"cityName"`
	DeviceId    string    `json:"deviceId"`
	Date        time.Time `json:"date"`
	Unit        string    `json:"unit"`
	Temperature float64   `json:"temperature"`
}

func NewMeasurementDto(cityName string, deviceId string, date time.Time, unit string, temperature float64) *MeasurementDto {
	return &MeasurementDto{
		CityName:    cityName,
		DeviceId:    deviceId,
		Date:        date,
		Unit:        unit,
		Temperature: temperature,
	}
}
